internal/autoevent/executor.go

```diff
@@ -75,7 +75,7 @@ func readResource(e *executor) (*dsModels.Event, common.AppError) {
 	vars[common.NameVar] = e.deviceName
 	vars[common.CommandVar] = e.autoEvent.Resource
 
-	evt, appErr := handler.CommandHandler(vars, "", common.GetCmdMethod, "")
+	evt, appErr := handler.CommandHandler(vars, "", common.GetCmdMethod, "", nil)
```

Comment:
Quickly looking at this, I think there may be a bug here. We are passing in `nil` for the `r *http.Request`. However, in the called function it never checks if the value is `nil` but performs some operations on `r`. For example:
https://github.com/edgexfoundry/device-sdk-go/blob/58d8c2fbfe57be80d06d3bf091d22c686d74c453/internal/handler/command.go#L488-L494

Reply:
-0 In revisiting this, one approach would be to change the signature of the following function (in `/device-sdk-go/internal/autoevent/executor.go`) as indicated below, which could then call `evt, appErr := handler.CommandHandler(vars, "", req)`
```
func readResource(e *executor, req *http.Request) (*dsModels.Event, common.AppError) {
...
...
	evt, appErr := handler.CommandHandler(vars, "", req)
...
...
}
```
- The challenge will be the call `evt, appErr := readResource(e, nil)` from _its_ calling function, which is as follows and, specifically, how to introduce and pass in a parameter of type `*http.Request`:
```
// Run triggers this Executor executes the handler for the resource periodically
func (e *executor) Run() { ... }
```